pkg/registry/core/event/strategy.go

```diff
@@ -78,7 +78,7 @@ func (eventStrategy) AllowCreateOnUpdate() bool {
 func (eventStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
 	groupVersion := requestGroupVersion(ctx)
 	event := obj.(*api.Event)
-	oldEvent := obj.(*api.Event)
+	oldEvent := old.(*api.Event)
```

Comment:
Good catch - can you please add some simple unit test verifying this?

Reply:
Yeah, I'll try to improve the test cases recently

Reply:
Friendly ping - will you be able to add some short unit test to it?

Reply:
I'd like to be able to contribute to the k8s community. 
I didn't commit the UT in time because I didn't know how to test this small modification; 
in addition, the large number of errors reported by ci actions made me a little helpless. 
I need some help.